Middleware/ConnectionKeepAlive.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Middleware;
+
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Connection as DriverConnection;
+use Doctrine\DBAL\Driver\Middleware;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+use Symfony\Bridge\Doctrine\ConnectionTimingWeakMap;
+
+class ConnectionKeepAlive implements Middleware, ConnectionNameAwareInterface
+{
+    private ConnectionTimingWeakMap $connectionTimingWeakMap;
+    private string $connectionName;
+
+    public function __construct(ConnectionTimingWeakMap $connectionTimingWeakMap, string $connectionName = 'default')
+    {
+        $this->connectionName = $connectionName;
+        $this->connectionTimingWeakMap = $connectionTimingWeakMap;
+    }
+
+    public function setConnectionName(string $name): void
+    {
+        $this->connectionName = $name;
+    }
+
+    public function wrap(Driver $driver): Driver
+    {
+        return new class($driver, $this->connectionTimingWeakMap, $this->connectionName) extends AbstractDriverMiddleware {
+            private Driver $driver;
+            private ConnectionTimingWeakMap $connectionTimingWeakMap;
+            private string $connectionName;
+
+            public function __construct(Driver $driver, ConnectionTimingWeakMap $connectionTimingWeakMap, string $connectionName = 'default')
+            {
+                $this->connectionName = $connectionName;
+                $this->connectionTimingWeakMap = $connectionTimingWeakMap;
+                $this->driver = $driver;
+
+                parent::__construct($driver);
+            }
+
+            public function connect(array $params): DriverConnection
+            {
+                $connection = parent::connect($params);
+                $this->connectionTimingWeakMap->setTimingPerConnection($connection, $this->connectionName);
```

Comment:
With this logic, we close the connection every 30s event if we rant queries in between.
What about decorating the DriverConnection to set the timing when any queries are sent?